libraries/c_sdk/standard/https/src/iot_https_client.c

```diff
@@ -0,0 +1,3326 @@
+    HTTPS_ON_NULL_ARG_GOTO_CLEANUP( pConnInfo->pNetworkInterface );
+    HTTPS_ON_NULL_ARG_GOTO_CLEANUP( pConnInfo->pAddress );
+    HTTPS_ON_ARG_ERROR_GOTO_CLEANUP( pConnInfo->addressLen > 0 );
+
+    /* Make sure the connection context can fit in the user buffer. */
+    HTTPS_ON_ARG_ERROR_MSG_GOTO_CLEANUP( pConnInfo->userBuffer.bufferLen >= connectionUserBufferMinimumSize,
+                                         IOT_HTTPS_INSUFFICIENT_MEMORY,
+                                         "Buffer size is too small to initialize the connection context. User buffer size: %d, required minimum size; %d.",
+                                         ( *pConnInfo ).userBuffer.bufferLen,
+                                         connectionUserBufferMinimumSize );
+
+    /* Make sure that the server address does not exceed the maximum permitted length. */
+    HTTPS_ON_ARG_ERROR_MSG_GOTO_CLEANUP( pConnInfo->addressLen <= IOT_HTTPS_MAX_HOST_NAME_LENGTH,
+                                         IOT_HTTPS_INVALID_PARAMETER,
+                                         "IotHttpsConnectionInfo_t.addressLen has a host name length %d that exceeds maximum length %d.",
+                                         pConnInfo->addressLen,
+                                         IOT_HTTPS_MAX_HOST_NAME_LENGTH );
+
+    /* Make sure that the ALPN protocols does not exceed the maximum permitted length. */
+    HTTPS_ON_ARG_ERROR_MSG_GOTO_CLEANUP( pConnInfo->alpnProtocolsLen <= IOT_HTTPS_MAX_ALPN_PROTOCOLS_LENGTH,
+                                         IOT_HTTPS_INVALID_PARAMETER,
+                                         "IotHttpsConnectionInfo_t.alpnProtocolsLen of %d exceeds the configured maximum protocol length %d. See IOT_HTTPS_MAX_ALPN_PROTOCOLS_LENGTH for more information.",
+                                         pConnInfo->alpnProtocolsLen,
+                                         IOT_HTTPS_MAX_ALPN_PROTOCOLS_LENGTH );
+
+    pHttpsConnection = ( _httpsConnection_t * ) ( pConnInfo->userBuffer.pBuffer );
+
+    /* Start with the disconnected state. */
+    pHttpsConnection->isConnected = false;
+
+    /* Initialize disconnection state keepers. */
+    pHttpsConnection->isDisconnecting = false;
+    pHttpsConnection->isDestroyed = false;
+
+    /* Initialize the queue of responses and requests. */
+    IotDeQueue_Create( &( pHttpsConnection->reqQ ) );
+    IotDeQueue_Create( &( pHttpsConnection->respQ ) );
+
+    /* This timeout is used to wait for a response on the connection as well as
+     * for the timeout for the connect operation. */
+    if( pConnInfo->timeout == 0 )
+    {
+        pHttpsConnection->timeout = IOT_HTTPS_RESPONSE_WAIT_MS;
+    }
+    else
+    {
+        pHttpsConnection->timeout = pConnInfo->timeout;
+    }
+
+    /* pNetworkInterface contains all the routines to be able to send/receive data on the network. */
+    pHttpsConnection->pNetworkInterface = pConnInfo->pNetworkInterface;
+
+    /* The address from the connection conf
```

Comment:
If both of these fields should include the NULL, then why not just compare that they're equal? This line is confusing.

Reply:
respHandle->readHeaderValueLength will not include a NULL.